modules/gis/src/lib/flatGeojson-to-binary-types.ts

```diff
@@ -0,0 +1,46 @@
+import type {TypedArray} from '@loaders.gl/schema';
+
+export type PropArrayConstructor =
+  | Float32ArrayConstructor
+  | Float64ArrayConstructor
+  | ArrayConstructor;
+
+export type Points = {
+  type: 'Point';
+  positions: Float32Array | Float64Array;
+  globalFeatureIds: Uint16Array | Uint32Array;
+  featureIds: Uint16Array | Uint32Array;
+  numericProps: {[key: string]: TypedArray};
+  properties: {}[];
+  fields: {
+    id?: string | number;
+  }[];
+};
+
+export type Lines = {
+  type: 'LineString';
+  positions: Float32Array | Float64Array;
+  pathIndices: Uint16Array | Uint32Array;
+  globalFeatureIds: Uint16Array | Uint32Array;
+  featureIds: Uint16Array | Uint32Array;
+  numericProps: {[key: string]: TypedArray};
+  properties: {}[];
+  fields: {
+    id?: string | number;
+  }[];
+};
+
+export type Polygons = {
+  type: 'Polygon';
+  positions: Float32Array | Float64Array;
+  polygonIndices: Uint16Array | Uint32Array;
+  primitivePolygonIndices: Uint16Array | Uint32Array;
+  triangles: number[];
```

Comment:
Is `triangles` new? Is that the output from `earcut`? (I might have missed some discussions about this)

Reply:
It has been present in the `MVTLoader` for some time now. It is indeed the output from earcut, and performing it in the loader means the costly operation can be done in the worker https://github.com/visgl/loaders.gl/pull/1356